client/me/developer/main.jsx

```diff
@@ -26,8 +28,63 @@ class Developer extends Component {
 		recordTracksEvent( 'calypso_me_is_dev_account_toggled', {
 			enabled: isDevAccount ? 1 : 0,
 		} );
+
+		const cookies = cookie.parse( document.cookie );
+
+		if ( isDevAccount && ! [ 'completed', 'dismissed' ].includes( cookies.developer_survey ) ) {
+			const noticeId = 'developer-survey';
+
+			this.props.successNotice(
+				this.props.translate(
+					'Your experience as a developer matters to us! Please share your thoughts by filling out {{surveyLink}}this quick survey{{/surveyLink}}.',
+					{
+						components: {
+							surveyLink: (
+								<a
+									href="handle_me"
+									target="_blank"
+									rel="noopener noreferrer"
+									onClick={ () => {
+										recordTracksEvent( 'calypso_me_developer_survey_clicked' );
+
+										this.props.removeNotice( noticeId );
+									} }
+								/>
+							),
+						},
+					}
+				),
+				{
+					id: noticeId,
+					isPersistent: true,
+					showDismiss: false,
+					button: this.props.translate( 'Dismiss' ),
+					onClick: () => {
+						recordTracksEvent( 'calypso_me_developer_survey_dismissed' );
+
+						document.cookie = cookie.serialize( 'developer_survey', 'dismissed', {
+							path: '/',
+							maxAge: 24 * 60 * 60, // 1 day
+						} );
+
+						this.props.removeNotice( noticeId );
+					},
+				}
+			);
+		}
 	};
 
+	componentDidMount() {
+		const urlParams = new URLSearchParams( window.location.search );
+
+		if ( urlParams.get( 'survey' ) === 'completed' ) {
+			document.cookie = cookie.serialize( 'developer_survey', 'completed', {
+				path: '/',
+				maxAge: 365 * 24 * 60 * 60, // 1 years
```

Comment:
I am not sure that 1 year is enough, WDYT?

Reply:
Unfortunatelly the limit is 400 days https://developer.chrome.com/blog/cookie-max-age-expires#:~:text=Ari%20Chivukula-,As%20of%20Chrome%20release%20M104%20(August%202022)%20cookies%20can%20no,when%20the%20browsing%20session%20ends.

Reply:
Ah, noted! Thanks for looking up the info! Setting the max age to 1 year sounds reasonable.